presto-main/src/main/java/io/prestosql/operator/scalar/StringFunctions.java

```diff
@@ -877,6 +880,74 @@ public static Slice toUtf8(@SqlType("varchar(x)") Slice slice)
         return slice;
     }
 
+    private static final String UTF8_DEFAULT_REPLACEMENT = "\uFFFD";
+
+    private static String stringFromCharset(Slice charset, Slice slice)
+    {
+        try {
+            return slice.toString(Charset.forName(charset.toStringUtf8()));
+        }
+        catch (UnsupportedCharsetException e) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Unsupported character set");
+        }
+        catch (IllegalCharsetNameException e) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Illegal character set");
+        }
+    }
+
+    @Description("Decodes string with a specific character set")
+    @ScalarFunction
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice fromCharset(@SqlType("varchar") Slice charset, @SqlType(StandardTypes.VARBINARY) Slice slice, @SqlType(StandardTypes.BIGINT) long replacementCodePoint)
```

Comment:
I don't think we need this variant that takes a replacement codepoint. I'm not sure why we added it for `from_utf8`, but I'd rather not add it unless we have a good reason. We can add it later if needed.

Reply:
I don't think we need it neither. Users can use `regexp_replace` to change from the default replacement to any characters they want.